app/main.py

```diff
@@ -2,5 +2,16 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    res = {}
+
+    def wrapper(*args, **kwargs) -> Callable:
+        key = (args, frozenset(kwargs.items()))
+        if key in res:
+            print("Getting from cache")
+            return res[key]
+        else:
+            print("Calculating new result")
+            res[key] = func(*args, **kwargs)
+            return res[key]
```

Comment:
You can rewrite this function with using one return. It will make your code more readable
```suggestion
        if key in res:
            print("Getting from cache")
        else:
            print("Calculating new result")
            res[key] = func(*args, **kwargs)
        return res[key]
```